src/sharechain/p2chain.rs

```diff
@@ -533,6 +534,41 @@ mod test {
         }
     }
 
+    #[test]
+    fn does_ot_set_tip_unless_full_chain() {
+        let mut chain = P2Chain::new_empty(10, 5);
+
+        let mut prev_hash = BlockHash::zero();
+        let mut tari_block = Block::new(BlockHeader::new(0), AggregateBody::empty());
+        for i in 1..5 {
+            tari_block.header.nonce = i;
+            let address = new_random_address();
+            let block = P2Block::builder()
+                .with_timestamp(EpochTime::now())
+                .with_height(i)
+                .with_tari_block(tari_block.clone())
+                .with_miner_wallet_address(address.clone())
+                .with_prev_hash(prev_hash)
+                .build();
+            prev_hash = block.generate_hash();
+            chain.add_block_to_chain(block.clone()).unwrap();
+        }
+        tari_block.header.nonce = 5;
+        let address = new_random_address();
+        let block = P2Block::builder()
+            .with_timestamp(EpochTime::now())
+            .with_height(5)
+            .with_tari_block(tari_block.clone())
+            .with_miner_wallet_address(address.clone())
+            .with_prev_hash(prev_hash)
+            .build();
+        prev_hash = block.generate_hash();
+        chain.add_block_to_chain(block.clone()).unwrap();
+
+        let level = chain.get_tip().unwrap();
+        assert_eq!(chain.get_tip().unwrap().height, 5);
```

Comment:
Please add a test where you just add block 5, it should not change the tip from 0